manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/ConnectorWorker.java

```diff
@@ -0,0 +1,242 @@
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.openshift.cloud.api.connector.models.Connector;
+import com.openshift.cloud.api.connector.models.ConnectorRequest;
+import com.openshift.cloud.api.connector.models.ConnectorState;
+import com.openshift.cloud.api.connector.models.ConnectorStatusStatus;
+import com.openshift.cloud.api.connector.models.DeploymentLocation;
+import com.openshift.cloud.api.connector.models.KafkaConnectionSettings;
+import com.openshift.cloud.api.connector.models.ServiceAccount;
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.RhoasService;
+import com.redhat.service.bridge.manager.connectors.ConnectorsApiClient;
+import com.redhat.service.bridge.manager.dao.ConnectorsDAO;
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+import com.redhat.service.bridge.rhoas.RhoasTopicAccessType;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+@ApplicationScoped
+public class ConnectorWorker extends AbstractWorker<ConnectorEntity> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConnectorWorker.class);
+
+    public static final String KAFKA_ID_IGNORED = "kafkaId-ignored";
+
+    @ConfigProperty(name = "managed-connectors.cluster.id")
+    String mcClusterId;
+
+    @ConfigProperty(name = "managed-connectors.kafka.bootstrap.servers")
+    String kafkaBootstrapServer;
+
+    @ConfigProperty(name = "managed-connectors.kafka.client.id")
+    String serviceAccountId;
+
+    @ConfigProperty(name = "managed-connectors.kafka.client.secret")
+    String serviceAccountSecret;
+
+    @Inject
+    ConnectorsDAO connectorsDAO;
+
+    @Inject
+    RhoasService rhoasService;
+
+    @Inject
+    ConnectorsApiClient connectorsApi;
+
+    @Override
+    protected PanacheRepositoryBase<ConnectorEntity, String> getDao() {
+        return connectorsDAO;
+    }
+
+    @Override
+    protected ConnectorEntity runCreateOfDependencies(ConnectorEntity connectorEntity) {
+        // This is idempotent as it gets overridden later depending on actual state
+        connectorEntity = setStatus(connectorEntity, ManagedResourceStatus.PROVISIONING);
+
+        // Step 1 - Create Kafka Topic
+        info(LOGGER,
+                String.format("Creating Kafka Topic for '%s' [%s]",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+        rhoasService.createTopicAndGrantAccessFor(connectorEntity.getTopicName(), RhoasTopicAccessType.PRODUCER);
+
+        // Step 2 - Create Connector
+        info(LOGGER,
+                String.format("Creating Managed Connector for '%s' [%s]",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+        Connector connector = connectorsApi.getConnector(connectorEntity);
```

Comment:
If I understood correctly we'll have just one Worker that models the whole provisioning and that is idempotent right? So we don't model each step in the DB indipendently

Reply:
Correct. If you recall we agreed to dispense of the sub-statuses. If none of the provisioning is asynchronous we can complete it within a single traversal of the applicable `Worker`. If however a stage is asynchronous (i.e. MC's) the `Worker` exits and processing recommenced in the next re-try.